### DISCUSSION TITLE
Creating an InMemoryDataset with HeteroData

### DISCUSSION THREAD

**Discussion by mdanb** [pyg-team/pytorch_geometric#7521] [Q&A] [answered]
I have `HeteroData` which I'm using in an `InMemoryDataset`. From the docs, I need to save the data in `process` as:

```      
data, slices = self.collate([data])
torch.save((data, slices), self.processed_paths[0])
```
However, it seems like `collate` doesn't work with `HeteroData`, only `Data`. So what's the best solution here?

**Comment by mdanb** [comment]
Actually, I was able to `collate` even though it expects `Data`

**Reply by rusty1s** [reply, marked as answer]
Yes, it should support both `Data` and `HeteroData`. Let me update the type hints.
